File: dotnet/Allmark/Render/HighlightRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class HighlightRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "highlight",
            Render = Render,
        };
    }

    public static void Render(MarkdownNode node, RendererState state, bool? decode = true)
    {
        RenderTag.Execute(node, state, "mark");
    }
}

File: zig/src/render/highlightRenderer.zig
const std = @import("std");
const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const RendererState = @import("../types/RendererState.zig").RendererState;
const Renderer = @import("../types/Renderer.zig").Renderer;
const renderTag = @import("renderTag.zig").renderTag;

pub fn render(node: *const MarkdownNode, state: *RendererState, decode: ?bool) void {
    _ = decode;
    renderTag(node, state, "mark", true);
}

pub const highlightRenderer = Renderer{
    .name = "highlight",
    .render = render,
};
